pkg/actions/karpenter/karpenter.go

```diff
@@ -45,6 +45,7 @@ func NewInstaller(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, stackManager
 		CreateServiceAccount: api.IsEnabled(cfg.Karpenter.CreateServiceAccount),
 		ClusterEndpoint:      cfg.Status.Endpoint,
 		Version:              cfg.Karpenter.Version,
+		CreateNamespace:      len(cfg.FargateProfiles) == 0,
```

Comment:
:thinking: whats this about? also missing a unit test

Reply:
No it's not. :)

Also, basically, when a fargate profiles is active, it will create the `karpenter` namespace, hence, we should tell helm not to create it itself.

Reply:
> No it's not. :)

I can't see a corresponding change in ` pkg/actions/karpenter/create_test.go `

Reply:
@aclevername Ah yes. The reason for that is that I'm using a fake Karpenter here and the arguments are just the context. Anything else is set up through `NewKarpenterInstaller` which I can't check here.

Meaning, I can't test that there, is why it's tested in here. 

I can try and add an integration test, but that requires a new cluster which would increase the runtime by another 20-30 minutes.

Reply:
I guess I can set up a real Karpenter installer with a fake helm installer and veryify that the fake helm installer was called with Namespace off.

Reply:
Actually, no. :D because I'm setting this up in `NewInstaller` which wires up the correct modules.

Reply:
Hmmm, maybe I can if I can make helm driver to be `in Memory` for testing. 🤔

Reply:
Nope, sorry. Can't do it. I thought maybe I can change the helm installer in between, but karpenter installer here is an interface.

What I can do is maybe shift to logic to testable place. Something like, inside Karpenter. But that didn't have access to the config so far, because it didn't needed it. I guess now it does. :)

Reply:
I've put the choosing logic into Karpenter itself, so now I can just pass in the whole config and test the fargate profiles len logic in there.

Reply:
@aclevername Okay, sorry for the long rants... How do you find the new changes? I shifted the config processing into Karpenter instead of the `New*` function.

Reply:
lgtm